File: CypherKeeperUI/CypherKeeper/src/app/Models/SavedKeyModel.ts
import { tbTwoFactorAuthModel_ADD } from "./tbTwoFactorAuthModel";

export interface SavedKeyModel {
    key: ToSaveKey;
    stringKeyFields: ToSaveStringKeyField[];
    twoFactorAuths: tbTwoFactorAuthModel_ADD[];
}

export interface ToSaveKey {
    id: string | null;
    parentGroupId: string;
    name: string;
    userName: string;
    password: string;
    websiteId: string | null;
    notes: string;
}

export interface ToSaveStringKeyField {
    id: string | null;
    name: string;
    value: string;
}

File: CypherKeeperAPI/CypherKeeper/CypherKeeper.Model/SavedKeyModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CypherKeeper.Model
{
    public class SavedKeyModel
    {
        public ToSaveKey Key { get; set; }
        public List<ToSaveStringKeyField> StringKeyFields { get; set; }
    }

    public class ToSaveKey
    {
        public Guid? Id { get; set; }
        public Guid ParentGroupId { get; set; }
        public String Name { get; set; }
        public String UserName { get; set; }
        public String Password { get; set; }
        public Guid? WebsiteId { get; set; }
        public String Notes { get; set; }
    }

    public class ToSaveStringKeyField
    {
        public Guid? Id { get; set; }
        public String Name { get; set; }
        public String Value { get; set; }
    }
}
